src/compiler/checker.ts

```diff
@@ -2766,7 +2766,9 @@ namespace ts {
         }
 
         function getBaseTypeNodeOfClass(type: InterfaceType): ExpressionWithTypeArguments {
-            return getClassExtendsHeritageClauseElement(<ClassLikeDeclaration>type.symbol.valueDeclaration);
+            let classDeclaration =
+                getDeclarationOfKind(type.symbol, SyntaxKind.ClassDeclaration) || type.symbol.valueDeclaration;

```

Comment:
I didn't think so either, but it fails for symbols with `kind === SyntaxKind.ClassExpression` (among others). At any rate @ahejlsberg suggested a better way -- keep valueDeclaration correct when assigning it in the binder and when merging symbols.